lib/matplotlib/widgets.py

```diff
@@ -2685,6 +2714,10 @@ def onselect(eclick: MouseEvent, erelease: MouseEvent)
 
     interactive = _api.deprecate_privatize_attribute("3.5")
 
+    maxdist = _api.deprecated("3.5")(
+        property(lambda self: self.grab_range)
+        )
```

Comment:
I don't think this will work as intended. `deprecated()` is a decorator that must be applied to a function. However a property is not callable, so I think the warning will not be emitted. Did you check this?

Also you need a replacement for the setter.

Something like this *may* work, please verify. I did not have time to check myself.

```suggestion
    maxdist = property(
        _api.deprecated("3.5", alternative='grab_range')(lambda self: self.grab_range)
        _api.deprecated("3.5", alternative='grab_range')(lambda: self, val: self.grab_range = val)
        )
```

Ping @anntzer is there a better way to rename an attribute?

Reply:
Actually `deprecated` has special handling for properties, and does work as is -- but you still need to provide a setter, so something like
```python
maxdist = _api.deprecated("3.5", name="maxdist", alternative="grab_range")(
    property(lambda self: self.grab_range,
             lambda self, value: setattr(self, "grab_range", value)))
```

Reply:
I did check that the property was working but forgot about the setter! This should be fixed in 8682b5f now.

Reply:
(side point: support for properties is explicitly listed at https://matplotlib.org/devdocs/api/_api_api.html#matplotlib._api.deprecation.deprecated)